proposed/2021/MachineReadableOutput.md

```diff
@@ -0,0 +1,807 @@
+   Top-level Package      Requested   Resolved   Reason(s)   Alternative
+   > NuGet.Core           2.13.0      2.13.0     Legacy
+
+   Transitive Package          Resolved              Reason(s)   Alternative
+   > NuGet.Packaging.Core      4.8.0-preview3.5278   Legacy      NuGet.Packaging >= 0.0.0
+
+#### `> dotnet list package --include-transitive --deprecated --framework net5.0 --format json`
+
+```json
+{
+  "version": 1,
+  "parameters": "-include-transitive --outdated --framework net5.0",
+   "sources": [
+    "https://api.nuget.org/v3/index.json",
+    "https://apidev.nugettest.org/v3-index/index.json"
+  ],
+  "projects": [
+    {
+      "path": "src/lib/MyProjectA.csproj",
+      "frameworks": [
+      ]
+    },
+    {
+      "path": "src/lib/MyProjectB.csproj",
+      "frameworks": [
+        {
+          "framework": "net5.0",
+          "topLevelPackages": [ 
+              {
+                "id": "NuGet.Core",
+                "requestedVersion": "2.13.0",
+                "resolvedVersion": "2.13.0",
+                "deprecationReasons": ["Legacy"]
+              }
+            ],
+            "transitivePackages": [
+              {
+                "id": "Microsoft.CSharp",
+                "resolvedVersion": "4.0.1",
+                "deprecationReasons": ["Legacy"],
+                "alternativePackage": {
+                  "id": "NuGet.Packaging",
+                  "versionRange": "[0.0.0,)"
+                }
+              }
+      ...
+            ]
+        }
+      ]
+    }
+  ]
+}
+```
+
+## Compatibility
+
+We start with `version 1`, as long as we don't remove or rename then it'll be backward compatible. In case [we change version](https://stackoverflow.com/a/13945074) just add new properties, keep old ones even it's not used.
+By default `--format json` will output latest schema version of json, but we can set older version of output like `--output-version 1`, intention behind is we don't want to customer CI build script because sdk/nuget version is upgraded and keep it predictable.
+
+## Out-of-scope
+
+* We won't support saving the machine-readable output to disk as part of this spec. The work-around is for the consumer to read from the console's stdout stream.
+* At this point, no other CLI commands (e.g. dotnet list reference) will be within scope for this feature.
+"--parsable" option needs separate spec.
+* Currently license info is not emitted from any cli command, it could be quite useful, we should consider in the future.
+
+## Rationale and alternatives
+Currently, no other `dotnet command` implemented this, this is the 1st time dotnet command implementing `json`(etc) output, so it could become example for others next time they implement.
+Please note, except "tab completion" (for dotnet) part all changes would be inside NuGet.Client repo(under NuGet.Core), and risk of introducing regression is low.(`--format text` refactoring related changes only come into my mind.), no impact on dotnet sdk.
```

Comment:
I appreciate the time and thought for tab-completion and SDK impact! And many kudos for bringing us in on the ground floor on this effort as well.